src/main/java/appeng/core/sync/packets/PacketIfaceTermUpdate.java

```diff
@@ -0,0 +1,432 @@
+package appeng.core.sync.packets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompressedStreamTools;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraftforge.common.util.Constants.NBT;
+
+import appeng.client.gui.implementations.GuiInterfaceTerminal;
+import appeng.core.sync.AppEngPacket;
+import appeng.core.sync.network.INetworkInfo;
+import appeng.helpers.Reflected;
+import cpw.mods.fml.common.network.ByteBufUtils;
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.buffer.Unpooled;
+
+/**
+ * Packet used for interface terminal updates. Packet allows the server to send an array of command packets, which are
+ * then processed in order. This allows for chaining commands to produce the desired update.
+ */
+public class PacketIfaceTermUpdate extends AppEngPacket {
+
+    public static final int CLEAR_ALL_BIT = 1;
+    public static final int DISCONNECT_BIT = 2;
+
+    private final List<PacketEntry> commands = new ArrayList<>();
+    private int statusFlags;
+
+    @Reflected
+    public PacketIfaceTermUpdate(final ByteBuf buf) throws IOException {
+        decode(buf);
+    }
+
+    public PacketIfaceTermUpdate() {}
+
+    private void decode(ByteBuf buf) {
+        this.statusFlags = buf.readByte();
+        int numEntries = buf.readInt();
+
+        for (int i = 0; i < numEntries; ++i) {
+            int type = buf.readByte();
+
+            try {
+                switch (type) {
+                    case 0 -> this.commands.add(new PacketAdd(buf));
```

Comment:
Flat magic numbers here can be replaced with enum match, from converted `int type` into PacketType